magento/pwa-studio
# Fix "Data Fetch Error" on product page in offline mode (#2489)

## Description
Initially I was thinking the wrong behavior ( check videoscreen below)  is caused by Apollo Retry Link and generally it is, but we have to change condition to handle it. 

### Scenario: 
1. Being online go to some product page
2. Switch browser to offline mode
3. Open the same product page (must be cached)
4. You see product page
5. In the background, you are able to see graphql retry requests (retry link)
6. After some tries you will see "Data Fetch Error" instead of product page

Check the current code: 
```js
    if (loading && !product) return fullPageLoadingIndicator;
    if (error) return <div>Data Fetch Error</div>;

    if (!product) {}
```

It works when you are online, but when you are offline this is not good, because when you got the `error` though you have product data then you won't display it, because of that error which is caused by retry link.

Being offline you have product data from apollo cache and you want to display it though you get errors from retrylink

So the change is small but needed for us because of PWA purposes. 


## Related Issue
Closes #2489 

## Acceptance 
Just go trough my scenario. 
### Verification Stakeholders
@sirugh 


### Verification Steps
1. Being online go to some product page
2. Switch browser to offline mode
3. Open the same product page (must be cached)
4. You see product page
5. In the background, you are able to see graphql retry requests (retry link)
6. After some tries you should see product page still.

## Screenshots / Screen Captures (if appropriate)
### Wrong behavior: 
![ezgif-4-2d2e07fceced](https://user-images.githubusercontent.com/7935392/84793332-40f70400-aff5-11ea-944f-d7ed08c8e7db.gif)

### Good behavior: 
![ezgif-4-a69f7dd662f9](https://user-images.githubusercontent.com/7935392/84793780-cc709500-aff5-11ea-843d-3c1980848643.gif)
